guides/updating/update-indy-sdk-to-askar.md

```diff
@@ -0,0 +1,168 @@
+
+:::
+
+:::caution
+
+Postgres is not supported, yet.
+
+:::
+
+:::caution
+
+The migration script is supported for 0.4.x (or moving from 0.3.x to 0.4.x).
+Please refer to [Migrating from AFJ 0.3.x to 0.4.x](./versions/0.3-to-0.4.md)
+to get to 0.4.x.
+
+:::
+
+## What does the migration do internally?
+
+The migration script does the following to make sure everything is migrated
+properly, and if anything goes wrong, it will revert back to a working state.
+
+### Create a backup
+
+Because undefined behaviour might occur, we create a backup in the new `tmp`
+directory from Aries Framework JavaScript. if some error occurs, it will be
+reverted back to the backed-up state and if no error occurs, it will delete the
+backup from the temporary directory.
+
+### Migrate the database to an Aries Askar structure
+
+The Indy-sdk and Aries Askar have different database structures. So first of
+all we need to change some table names, decrypt all the items with the old Indy
+keys, encrypt the items with the new Aries Askar keys and store them inside the
+new structure.
+
+### Try to open the wallet in the new Aries Askar structure
+
+When the wallet is correctly transformed, the wallet will be attempted to be
+openend.
+
+### Update the keys
+
+Aries Askar has a specific way to store keys and every key, defined by the
+category of `Indy::Key` will be migrated.
+
+### Update the credential definitions
+
+:::danger
+
+Migration of credential definitions is not yet supported. This is why it is
+strongly advised not to run this script in a node.js environment.
+
+:::
+
+### Update the link secret(s) (master secret)
+
+The link sercets, identified by the category `Indy::MasterSecret`, are updated
+next. They are stored inside a new `AnonCredsLinkSecretRecord`.
+
+:::caution
+
+Since we have to set a default link secret, some additional logic is added to
+detect this. You can either supply a `defaultLinkSercetId` as a constructor
+parameter or it will be based on your `walletId`.
+
+If there is no Indy record with the `defaultLinkSecretId` or the `walletId`, an
+error will be thrown and the migration will be restored.
+
+:::
+
+### Update the credentials
+
+The credentials, identified by the category `Indy::Credential` are updated
+last. They are stored in the new `AnonCredsCredentialRecord` as a one-to-one
+copy. This means that they now support more tags and will make querying a lot
+easier.
+
+### All the other records
+
+All the other records will be transferred without any updates as they are not
+Indy specific.
+
+## How to update
+
+Updating does not require a lot of code, but must be done with caution.
+
+### add the required dependencies:
+
+```sh
+yarn add @hyperledger/aries-askar-react-native @aries-framework/indy-sdk-to-askar-migration react-native-file-system
+```
+
+Below is the minimal code required for the migration to work. It is recommended
+to turn the logger on as it gives a lot of information regarding the migration.
```

Comment:
I think it would be good to mention this script only has to be run once, and it needs to be run after updating afj to 0.4.0 but before running the update assistant.

Also i think we should recommend users to store whether the agent is updated to askar on e.g. async storage or mmkv, so they can check it before launching the agent. Abit similar to how updating works (see the guide on updating)

Reply:
yes, good point! Will add a section and refer to guide on updating.